File: leetcode/easy/0226. Invert Binary Tree/main.ts
// Solved by @kitanoyoru
// https://leetcode.com/problems/invert-binary-tree/

/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

const helper = (root: TreeNode | null) => {
  if (!root) {
    return
  }

  let temp = root.left
  root.left = root.right
  root.right = temp

  helper(root.left)
  helper(root.right)
}

const invertTree = (root: TreeNode | null): TreeNode | null => {
  helper(root)

  return root
}

File: leetcode/easy/0226. Invert Binary Tree/main.java
// Solved by @kitanoyoru
// https://leetcode.com/problems/invert-binary-tree/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
  private void helper(TreeNode root) {
    if (root == null) {
      return;
    }

    var temp = root.left;
    root.left = root.right;
    root.right = temp;

    helper(root.left);
    helper(root.right);
  }

  public TreeNode invertTree(TreeNode root) {
    this.helper(root);

    return root;
  }
}
